components/hab/src/command/origin/create.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{api_client::{self,
+                         Client},
+            common::ui::{Status,
+                         UIWriter,
+                         UI},
+            error::{Error,
+                    Result},
+            PRODUCT,
+            VERSION};
+use hyper::status::StatusCode;
+
+pub fn start(ui: &mut UI, bldr_url: &str, token: &str, origin: &str) -> Result<()> {
+    let api_client = Client::new(bldr_url, PRODUCT, VERSION, None).map_err(Error::APIClient)?;
+
+    ui.status(Status::Creating, format!("origin {}.", origin))?;
+
+    match api_client.create_origin(origin, token) {
+        Ok(_) => {
+            ui.status(Status::Created, format!("origin {}.", origin))
+              .map_err(Error::from)
+        }
+        Err(api_client::Error::APIError(StatusCode::Conflict, msg)) => {
+            ui.fatal(format!("Unable to create origin {}", origin))?;
+            ui.fatal("Origins may only be created if they don't already exist")?;
```

Comment:
It would probably be good to be consistent with punctuation/line endings for these messages and include periods at the end.  Similarly with the error messages below.